libs/cardano-ledger-binary/testlib/Test/Cardano/Ledger/Binary/RoundTrip.hs

```diff
@@ -104,10 +106,18 @@ roundTripExpectation ::
   t ->
   Expectation
 roundTripExpectation trip t =
-  forM_ [natVersion @2 .. maxBound] $ \version ->
-    case roundTrip version trip t of
-      Left err -> expectationFailure $ "Failed to deserialize encoded: " ++ show err
-      Right tDecoded -> tDecoded `shouldBe` t
+  forM_ [natVersion @2 .. maxBound] (roundTripExpectation' trip t)
+
+roundTripExpectation' ::
+  (Show t, Eq t, Typeable t, HasCallStack) =>
+  Trip t t ->
+  t ->
+  Version ->
+  Expectation
+roundTripExpectation' trip t version =
```

Comment:
Please, no ticks :pray: 

```suggestion
-- | Test roundtripping with exact version
roundTripExactExpectation ::
  (Show t, Eq t, Typeable t, HasCallStack) =>
  Trip t t ->
  t ->
  Version ->
  Expectation
roundTripExactExpectation trip t version =
```

Reply:
yea, I guess it's a bad habit. `roundTripExactExpectation` is much clearer :)